.github/workflows/generate-post-release-pr.yml

```diff
@@ -38,7 +38,11 @@ jobs:
             minor="${BASH_REMATCH[2]}"
             patch="${BASH_REMATCH[3]}"
             
-            two_releases_ago="$major.$((minor - 1)).$patch"
+            if [[ "$patch" == "0" ]]; then
+              two_releases_ago="$major.$((minor - 1)).$patch"
```

Comment:
I'm not sure this is always correct since it will always be `$major.$((minor - 1)).0` and so will miss any patches that were made on `$major.$((minor - 1)).x`

I went a slightly different route by embedding the post-release workflow directly into release workflow which helps with this particular problem: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/805ce0a3cc73cf0d059a870cd0bb88b9edac69cc/.github/workflows/release.yml#L239

but it hasn't run successfully yet(!) so not necessarily recommending to copy it

Reply:
> I'm not sure this is always correct since it will always be $major.$((minor - 1)).0 and so will miss any patches that were made on $major.$((minor - 1)).x

Yeah I think you're right. If most recent versions are `1.43.2` and `1.44.0`, then `two_releases_ago` resolves to `1.43.0` when it should resolve to `1.43.2`.

Reply:
By this same logic, I believe this logic for calculating the `PRIOR_VERSION` is also not always correct: https://github.com/open-telemetry/opentelemetry-java/blob/main/.github/workflows/release.yml#L37-L58

I think the difference is that it doesn't really matter that `PRIOR_VERSION` is sometimes incorrect in `release.yml` based on where how its used.

To really be correct always, i think we need a list of release tags, ordered by the time they were publish. From such a list, we could fetch the most recent, 2nd most recent, n-th most recent release.